crates/ruff_db/Cargo.toml

```diff
@@ -20,6 +20,7 @@ camino = { workspace = true }
 countme = { workspace = true }
 dashmap = { workspace = true }
 filetime = { workspace = true }
+itertools = { workspace = true }
```

Comment:
argh nooo :(

Reply:
Not all itertools are equal, though, right? ;) I checked out their implementation for `Iterator::join` and it looks pretty reasonable to me. But I might get rid of this again as a result of addressing your other comment.

Reply:
Yeah, it just feels like a dependency that we could possibly remove by vendoring the few methods that we need (maybe join?)

Reply:
I, in general, want to remove the number of Ruff's dependencies. That's why I'm a bit annoying about introducing new dependencies ;)

Reply:
Yeah I totally get that. I think removing the `itertools` dependency might be a somewhat large task as we already use it quite a lot in the linter crate. But I'm totally on board with thinking it through carefully when we add new dependencies.

I've got rid of it from this PR, anyway :-)